### DISCUSSION TITLE
Connecting to VPNGate requires compat-mode

### DISCUSSION THREAD

**Discussion by devcoder00** [schwabe/ics-openvpn#1603] [Configuration problems] [answered]
i am facing this issues : https://github.com/schwabe/ics-openvpn/issues/612

exactly same issues ? what solution ? your old fix doesn't work.

can you please help me out. i would like to pay amount for help or maintenance project.

it not connecting showing " wait 2s seconds between connection attempt" in notification ? what solution for this ? bruh?

**Comment by schwabe** [comment]
You need to look into your log to understand why openvpn is not connecting. That message only says that the app will wait 2s before trying to connect again.

**Reply by devcoder00** [reply]
"That message only says that the app will wait 2s before trying to connect again."
but it doesn't connect time goes keep increasing. ? why?. the config file is correct

**Reply by schwabe** [reply]
I don't have a crystal ball. WIthout you sharing or understand your log file yourself, there is no way to know what is wrong. Your config file might work with old OpenVPN version but might require insecure settings/crypto that is no longer enabled by default.

**Reply by devcoder00** [reply]
> I don't have a crystal ball. WIthout you sharing or understand your log file yourself, there is no way to know what is wrong. Your config file might work with old OpenVPN version but might require insecure settings/crypto that is no longer enabled by default.


i will share my log before that can i know what is, "  Your config file might work with old OpenVPN version but might require insecure settings/crypto that is no longer enabled by default. "

what is this? i was having old version of your app and i suddenly upgrade and it not working ?.

maybe this can be a issues ? insecure settings/crypto that is no longer enabled by default how this enable  this ?

**Reply by schwabe** [reply]
See this document about all the Changes that were in OpenVPN 2.6:

https://github.com/OpenVPN/openvpn/blob/master/Changes.rst

If your config "suddenly" stops working, the OpenVPN server is older than 6-7 year or configured like such an old server.

**Reply by devcoder00** [reply]
so using 2.5 version gonna fix the issue right?.

i mostly use the files from https://www.vpngate.net/en/

**Reply by schwabe** [reply]
you can try setting compatiblity to openvpn 2.5.0 or 2.4.0 or even 2.3.0 in the basic settings of the VPN. My app will not allow you to use OpenVPN 2.5 instead of OpenVPN 2.6

**Reply by devcoder00** [reply]
> you can try setting compatiblity to openvpn 2.5.0 or 2.4.0 or even 2.3.0 in the basic settings of the VPN.
In your app? i only have your app source 

> My app will not allow you to use OpenVPN 2.5 instead of OpenVPN 2.6
i mean if download the old version from https://github.com/schwabe/ics-openvpn/releases of your app.

can you help me out?

if  newer version doesn't support old files then where i can get new files which can work without issues . like vpngate ? this site provide files for free.

OR
if i want use old version
 i mean using old source from https://github.com/schwabe/ics-openvpn/releases of your app which can work ?

**Reply by schwabe** [reply]
Old versions come with outdated insecure OpenSSL versions and I do not support them. But I am not the support team of your VPN Provider, please contact them if you cannot resolve the issue yourself.

**Reply by devcoder00** [reply]
hey? 

from where i can access this these options ? From where i can see these  compatibility mode option in this app source any specific class ?
![photo1680337341](https://user-images.githubusercontent.com/129509477/229274972-ceb69d63-a1ca-47f5-8808-70bba2be37d8.jpeg)

**Reply by schwabe** [reply]
This options sets `compat-mode`.

**Reply by devcoder00** [reply]
i can do my self but can you specify which class have compat-mode? and where i can find because  in this 
main/src/main/java/de/blinkt/OpenVpn/ i didn't get anything which is related or active compat mode as you told.

**Reply by devcoder00** [reply]
i saw the app source and there is non-thing which exists named as compat-mode

**Reply by devcoder00** [reply]
hello???

**Reply by schwabe** [reply]
I am not sure what you expect of me here. This is my spare time project and I do not enjoy helping people build other apps on top of my source code for free since I had very bad experience doing that in the past.

**Reply by devcoder00** [reply]
> I am not sure what you expect of me here. This is my spare time project and I do not enjoy helping people build other apps on top of my source code for free since I had very bad experience doing that in the past.

i not expecting anything, but just want to know, how to turn on compat mode. i build app of this source code and my files not working because compat mode is off. i want it to on. 
i never said that i am creating a third party app using this app. it is an app not any library how can we use this? i dont have any clues 

i running the build apk from this source on my Android os (PC one ).  so again i just want to know, how to turn on compat mode in this code.

it just for my OS Only. even i will donate to you using paypal for this app and help. pls just help me bit. 

there is non thing which is illegal that i am doing, which you meant by " people build other apps on top of my source code for free".

i dont  do these thing.  so you cn help me a bit!

**Reply by schwabe** [reply]
This is really an app and not a library to build other apps. And yes you might not be something bad but I completely lost all motivation to help others. The motivation of spending multiple hour of modifying an app just for yourself does not really make sense. And others how claimed similar things then suddenly had some commercial VPN string in the logs/code they shared with me even though they claimed it was just personal. Sorry, but there just too many bad Apples here.

**Reply by devcoder00** [reply]
> This is really an app and not a library to build other apps. And yes you might not be something bad but I completely lost all motivation to help others. The motivation of spending multiple hour of modifying an app just for yourself does not really make sense. And others how claimed similar things then suddenly had some commercial VPN string in the logs/code they shared with me even though they claimed it was just personal. Sorry, but there just too many bad Apples here.

i can't do anything about bad apples :( but some apples are left  good.

anyways, i understand people are selfish we cannot stop them. 

i always use your app but just making function to turn on always. if you want i can post the source on github under GPL. there will be no problem from my side.

and i can't do anything about motivation, i am actually sorry about that ! as this project is maintain by you which is very hard.

**Reply by devcoder00** [reply]
still @schwabe  do you have any plan to help ? if people copy your work then why you publish this on github ? why it is open source ? make it private.  private repo is answer to your rudeness  for common people or stop people to use it simple

**Reply by schwabe** [reply, marked as answer]
I think publishing it as open source is good thing and I think people should use it. But, they should use it under the terms of the GPL and they should either contribute in positive and meaningful way or leave me alone. I just very fed up with people that basically try to exploit me. I started this with good faith and help people in doubt but there are only so many times you do and being taken advantage of. On the other hand there are projects like eduVPN that actually respect the GPL and are constructive that I support. 

Also I am publishing this source code for free. That does not mean it comes with free support. If you really want support for this you either have to get on the good side with me or pay me for it. After all I am software developer and doing consulting and freelancing for other is not something I am doing for free. 

Especially with the "I want to build an app based on your app" I have been burned so many times that I just have no interest that anymore. Nobody that I know of spends weeks of programming for a small app for just some friends. That just sounds sketchy and has always turned out to be outright lying.

**Reply by devcoder00** [reply]
`**they should use it under the terms of the GPL and they should either contribute in positive and meaningful way or leave me alone.**`
did i said i will not ? of course i will do 
` **If you really want support for this you either have to get on the good side with me or pay me for it. After all I am software developer and doing consulting and freelancing for other is not something I am doing for free.**`
i already said i will pay 100$ using paypal only  because i have that paypal for international money transfer.

atleast help
here or telegram

**Reply by schwabe** [reply]
Closing this discussion as this leads nowhere and I already explained why do I spend my spare time with support for third party app building.
